pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,169 @@
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />
+            </FormGroup>
+        </Form>
+    );
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No group name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores.");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+function validate_group(id, groups) {
+    if (!id)
+        return _("No ID specified");
+
+    const id_num = parseInt(id);
+    if (id_num.toString() !== id || id_num < 0)
+        return _("The group ID must be positive integer.");
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].gid == id_num)
+            return _("A group with this ID already exists");
+    }
+
+    return null;
+}
+
+export function group_create_dialog(groups, setGroupsCardExpanded) {
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
+                const gids = groups
+                        .filter(g => g.name !== 'nobody')
+                        .map(group => group.gid);
+                const min_gid = parseInt(logindef.match(/^GID_MIN\s+(\d+)/m)[1] || 100);
+                const max_gid = parseInt(logindef.match(/^GID_MAX\s+(\d+)/m)[1] || 65534);
+
+                change("id", (Math.max(min_gid, Math.max(...gids.filter(id => id != max_gid)) + 1) + 1).toString());
+            });
+
+    let dlg = null;
+    const state = {
+        name: "",
+        id: "",
+    };
+    let errors = { };
+
+    function change(field, value) {
+        state[field] = value;
+        errors = { };
+
+        update();
+    }
+
+    function validate(name, id) {
+        const errs = { };
+
+        errs.name = validate_name(name, groups);
+        errs.id = validate_group(id, groups);
+        errors = errs;
+
+        return !has_errors(errs);
+    }
+
+    function create(name, id) {
+        const valid = validate(name, id);
+        if (valid) {
+            const group_add_cmd = ["/usr/sbin/groupadd", name, "-g", id];
```

Comment:
No hardcoded paths, please, they can cause trouble across different OSes. We have `$PATH` for that. I know we currently hardcode paths in a lot of places, but we need to stop and fix that.